docs/DesignMDIngestion.md

```diff
@@ -0,0 +1,126 @@
+# Metadata Ingestion from RING to Zenko
+
+## Description
+
+This is the design document discussing the ingestion process of existing metadata
+from the RING into Zenko with MongoDB.
+
+The primary use case is to copy the existing metadata from the RING/S3 Connector
+and ingest into the MongoDB backend that is used with Zenko so that users can
+connect existing buckets on RING to be managed by Zenko. This would allow users
+to complete operations such as metadata search and CRR. The ingestion process will
+also enable out-of-band updates. This will allow operations made directly to the RING
+to update and be reflected on Zenko. 
+
+This specific development will allow Zenko instances to copy and ingest existing
+metadata, copying the information from the raft logs completely and syncing MongoDB
+to be a parallel metadata database.
+
+## Purpose
+
+The primary purpose is to copy all existing metadata from pre-existing storage solutions
+to MongoDB in Zenko. The imported metadata will allow users to manage the bucket
+and search the metadata that is stored in MongoDB.
+
+## Design
+
+The ingestion process would be initiated bucket by bucket, as specified by the user.
+
+The proposed work flow will be as follows:
+* A RING bucket will be added as a storage location. An ingestion workflow will be
+  applied to a bucket created by the user using the created storage location, and
+  the option for `bucketMatch` will be true. This will allow 1 to 1 mapping between
+  the bucket defined through Zenko and the RING bucket.
+* Upon initiating the ingestion workflow:
+    * Record the last sequence ID from the Metadata Server (send a query using the
+      `metadataWrapper`) that is part of the S3 connector - this will serve as a
+      marker that will be stored as a `log offset` to bookmark the point between
+      logs that existed prior to the start of the Zenko instance, and the point
+      where new logs are added during the process where the old logs are copied.
+    * From backbeat, we will use the S3 protocols to send the following requests
+      to the S3 connector: 
+        * For the specified bucket, list all objects in the buckets. This will return
+          info such as the `object key`, and if `versioning` is enabled, `version id`.
+      Using the list of object keys, send a query directly to the metadata server
+      with the `metadataWrapper` class in Arsenal.
+        * This will get the JSON object for each object, which is put into kafka.
+
+With this design, Backbeat will include a new populator which reads from the Metadata
+server and queues the logs to Kafka. The consumer will not need to be changed.
+
+On deployment of Zenko, we will have 1 ingestion populator pod and 5 ingestion processors
```

Comment:
I say 3 or 5 processors, and can also be adjusted by the user

Reply:
Just to double check: 3 is the default right? And we generally deploy with 3 or 5

Reply:
ah actually its based on node count

Reply:
Both ingestion populator pod and ingestion processors willl be based on the node count?

Reply:
Only the processor. Populator will always just be one pod

Reply:
Cool, thank you!